### ISSUE TITLE
[BUG] input text field goes empty after filling in firefox

### ISSUE THREAD

**Issue by kemsky** [microsoft/playwright#21110]
**Context:**
- Playwright Version: 1.31.0
- Operating System: All
- Node.js version: v16.18.1
- Browser: Firefox

**Code Snippet**

Code is a simple as it can be, we are using `page.fill` and custom selectors.

```javascript
await this.page.fill(`${this.automation} >> auto=FormTextComponent||[name="${name}"] >> css=input`, text);
```

**Describe the bug**

Visually, it is exactly the same bug as closed #18121.

The following happens in our case: playwright is fast and it starts typing as soon as target input appears on the page, milliseconds later input is covered by transparent div aka loading indicator, playwright continues typing till the last character then it waits because (my guess) form controls are blocked by invisible div, input stays filled, once loading indicator is removed - input is cleared. Looks like some compensation action is implemented. There is no such issue in Chrome, but it suddenly stopped working in Windows Server 2012 and we had to switch to Firefox.

-----
Adding `page.click` before `page.fill` fixes issue. Very similar to https://github.com/microsoft/playwright/issues/20945#issuecomment-1436741089.

**Comment by yury-s** [comment]
> The following happens in our case: playwright is fast and it starts typing as soon as target input appears on the page, milliseconds later input is covered by transparent div aka loading indicator,

This is a racy behavior, you should probably wait until the overlay appears if it interferes with typing and only call fill after that. Alternatively you can pass [force](https://playwright.dev/docs/api/class-locator#locator-fill-option-force) to bypass some checks but this will likely just mask the issue.

> Code is a simple as it can be, we are using page.fill and custom selectors.

Can you please share full repro? we cannot act on this before we can reproduce it locally.

> Adding page.click before page.fill fixes issue. 

Do you have an idea why?

**Comment by kemsky** [comment]
@yury-s, it is racy, true. However, it is not so obvious:

- strictly speaking it is not a race to the user of UI because time-frame is very very short, literally there is one JS tick delay
- behavior is not consistent across browsers, in Chrome it just works
- behavior is bug-like - you see how input is filled and then it is blank again

So at least it should throw error because in fact it failed to fill input but instead test is failing because input is empty and form does not pass validation.

> Can you please share full repro? we cannot act on this before we can reproduce it locally.

I will try, but I also described in detail how to reproduce it, `setTimeout(<add div to cover input>, 0)` should work.

> Do you have an idea why?

I think `page.click` either adds at least one tick delay that is enough in this case or adds some internal checks according to https://playwright.dev/docs/actionability.

**Comment by yury-s** [comment]
> So at least it should throw error because in fact it failed to fill input but instead test is failing because input is empty and form does not pass validation, alternatively fill should succeed if it started typing successfully. 

According to your description the input text is successfully filled in but next moment the input is cleared for some reason (I suspect some page's logic). It may well be browser-specific behavior. We need a reproducible example which we could run locally to make this report actionable.

**Comment by yury-s** [comment]
We need a reproducible example that we could run locally to make this report actionable. Please file a new issue with a repro and link it to this one.
